### PR TITLE
network: do not allow to request invalid block count

### PR DESCRIPTION
**Pull request by AnnaShaleva**
The problem is in peer disconnection due to invalid GetBlockByIndex payload (the logs are from some patched neo-go version):
```
дек 15 16:02:39 glagoli neo-go[928530]: 2022-12-15T16:02:39.490Z        INFO        new peer connected        {"addr": "10.78.69.115:50846", "peerCount": 3}
дек 15 16:02:39 glagoli neo-go[928530]: 2022-12-15T16:02:39.490Z        WARN        peer disconnected        {"addr": "10.78.69.115:50846", "error": "invalid block count", "peerCount": 2}
дек 15 16:02:39 glagoli neo-go[928530]: 2022-12-15T16:02:39.490Z        INFO        started protocol        {"addr": "10.78.69.115:50846", "userAgent": "/NEO-GO:1.0.0/", "startHeight": 0, "id": 1339571820}
дек 15 16:02:39 glagoli neo-go[928530]: 2022-12-15T16:02:39.491Z        INFO        new peer connected        {"addr": "10.78.69.115:50856", "peerCount": 3}
дек 15 16:02:39 glagoli neo-go[928530]: 2022-12-15T16:02:39.492Z        WARN        peer disconnected        {"addr": "10.78.69.115:50856", "error": "invalid block count", "peerCount": 2}
дек 15 16:02:39 glagoli neo-go[928530]: 2022-12-15T16:02:39.492Z        INFO        started protocol        {"addr": "10.78.69.115:50856", "userAgent": "/NEO-GO:1.0.0/", "startHeight": 0, "id": 1339571820}
дек 15 16:02:39 glagoli neo-go[928530]: 2022-12-15T16:02:39.492Z        INFO        new peer connected        {"addr": "10.78.69.115:50858", "peerCount": 3}
дек 15 16:02:39 glagoli neo-go[928530]: 2022-12-15T16:02:39.493Z        INFO        started protocol        {"addr": "10.78.69.115:50858", "userAgent": "/NEO-GO:1.0.0/", "startHeight": 0, "id": 1339571820}
дек 15 16:02:39 glagoli neo-go[928530]: 2022-12-15T16:02:39.493Z        WARN        peer disconnected        {"addr": "10.78.69.115:50858", "error": "invalid block count", "peerCount": 2}
дек 15 16:02:39 glagoli neo-go[928530]: 2022-12-15T16:02:39.494Z        INFO        new peer connected        {"addr": "10.78.69.115:50874", "peerCount": 3}
дек 15 16:02:39 glagoli neo-go[928530]: 2022-12-15T16:02:39.494Z        INFO        started protocol        {"addr": "10.78.69.115:50874", "userAgent": "/NEO-GO:1.0.0/", "startHeight": 0, "id": 1339571820}
дек 15 16:02:39 glagoli neo-go[928530]: 2022-12-15T16:02:39.494Z        WARN        peer disconnected        {"addr": "10.78.69.115:50874", "error": "invalid block count", "peerCount": 2}
```

GetBlockByIndex payload can't be decoded, and the only possible cause is zero (or <-1, but it's probably not the case) block count requested.

Error is improved as far.

UPD: `0` blocks were requested:
```
дек 15 17:44:32 az neo-go[998229]: 2022-12-15T17:44:32.814Z        WARN        peer disconnected        {"addr": "10.78.69.115:55634", "error": "invalid block count: 0", "peerCount": 2}
```

### CODE REVIEW

File: pkg/network/server.go [hunk 1]
```diff
@@ -1252,7 +1252,7 @@ func (s *Server) requestBlocks(bq Blockqueuer, p Peer) error {
 	lq := s.bQueue.lastQueued()
 	if lq > pl.IndexStart {
```
Comment (roman-khimov) [inline pkg/network/server.go:1253]: `>=` here? We already have `pl.IndexStart` if they're equal and we should be requesting the next one.

File: pkg/network/server.go [hunk 2]
```diff
@@ -1252,7 +1252,7 @@ func (s *Server) requestBlocks(bq Blockqueuer, p Peer) error {
...
 		c := int16(h + blockCacheSize - lq)
-		if c < payload.MaxHashesCount {
+		if c < payload.MaxHashesCount && c > 0 {
```
Comment (roman-khimov) [inline pkg/network/server.go:1255]: This is not correct, if we have `c <= 0` then we're likely in a situation where some time has passed between `bq.BlockHeight()` and `s.bQueue.lastQueued()`, so we have an old height (0, for example) with some new last queued (4000, for example, 2000 blocks consumed already). This means that we should be doing a bit different thing here, we need both `lq` and the capacity left in the queue (`blockCacheSize - bq.len`) at the same time returned from the block queue. If the capacity is zero, we can't do a lot and no package should be sent at all, if it's less than `MaxHashesCount` then `pl.Count` should be adjusted for it.

Comment (AnnaShaleva) [reply pkg/network/server.go:1255]: > the only possible cause is zero (or <-1, but it's probably not the case)

Just for information, we've checked it, payload with `0` blocks count was requested in this case:
```
дек 15 17:44:32 az neo-go[998229]: 2022-12-15T17:44:32.814Z        WARN        peer disconnected        {"addr": "10.78.69.115:55634", "error": "invalid block count: 0", "peerCount": 2}
```

Comment (AnnaShaleva) [reply pkg/network/server.go]: Fixed.

File: pkg/network/server.go (hunk 1 above)
Comment (AnnaShaleva) [reply pkg/network/server.go]: Fixed.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**